### ISSUE TITLE
Make tarantool internal functions to call error with caller

### ISSUE THREAD

**Issue by Mons** [tarantool/tarantool#2399]
Now, if we call, for example
```lua
box.begin()
box.begin()
```
We got an exception from box.begin:
```
Operation is not permitted when there is an active transaction
```
Without line number.

Rather it would be:

```
test.lua:4: Operation is not permitted when there is an active transaction
```

**Comment by kostja** [comment]
This is impossible. LuaJIT has tail recursion elimination which leaves no trace of the original frame on the interpreter stack.

**Comment by rtsisyk** [comment]
It seems that `error(x, 2)` trick doesn't work in our case. All Lua API is implemented via FFI or Lua/C. C API creates a `struct error` object for every error, sets it into a global errno-like variable (struct diag) and returns -1. Both FFI and Lua/C calls C API, checks for -1, gets precreated `struct error` object and re-throws it to the upper levels as CDATA. Upper levels (IPROTO_CALL, IPROTO_EVAL, console, triggers, fiber.create()) extracts original `struct error` from Lua and saves back to our errno (struct diag). In this scheme any information added by Lua is simple discarded.

This problem can only be solved by adding [tracebacks](https://github.com/tarantool/tarantool/issues/677) to `struct error`. I've updated https://github.com/tarantool/tarantool/issues/677 and attached code we made during investigation of this ticket.
 
Duplicate of #677
